desktop-widgets/tab-widgets/TabDiveInformation.cpp

```diff
@@ -93,9 +93,12 @@ void TabDiveInformation::closeWarning()
 }
 
 void TabDiveInformation::updateWaterTypeWidget()
-{    // Decide on whether to show the water type/salinity combobox or not
-	if (prefs.salinityEditDefault || manualDive)
-	{         // if the preference setting has been checked or this is a manually-entered dive
+{
+    struct divecomputer *currentdc = get_dive_dc(current_dive, dc_number);
+    bool hasDCSalinity = currentdc && currentdc->salinity != 0;
```

Comment:
This will lead to weird results for dives with multiple dive computers (which is common in technical diving), where only some of the computers report salinity:

![image](https://user-images.githubusercontent.com/4742747/228684468-41a3ed16-f60a-4ae0-98b7-c4ca21309032.png)
(full disclosure: this import was fudged, the OSTC4 _does_ report salinity)

![image](https://user-images.githubusercontent.com/4742747/228684490-42313c5e-c7a5-4a20-8398-37cd1f8afe18.png)

So a better solution will be to check all dive computers used on the dive, and only allow editing if none of them report salinity.

Reply:
About is:
https://github.com/subsurface/subsurface/blob/eebb47ec22aafa34e29cd0385853fde10fe208ab/desktop-widgets/tab-widgets/TabDiveInformation.cpp#L182-L193

What would be "dc_value" when more than 1 DC?

Reply:
Correcting... Has already a merge between DCs salinity...
https://github.com/subsurface/subsurface/blob/eebb47ec22aafa34e29cd0385853fde10fe208ab/core/dive.c#L824-L839

We can use dive->salinity instead dc->salinity. What do you think?

Reply:
I am not sure that showing the user an average between the configured salinities on different dive computers is the right approach here. Using this average as the base for calculations may make sense, but showing it to the user is probably more confusing than helpful.
Salinity is different from measured dimensions like temperature or pressure - for these, differences between dive computers on the same dive are likely down to sensor tolerances, so it makes sense to use an average. For salinity, the user will always have set his dive computer(s) to a specific salinity, and if they differ between computers then there is a reason for it (potentially configuration error) - and not that they want to use an average.
So I think if we display a calculated average for salinity here we should at least use some form of indication (maybe a warning) to indicate to the user that this is happening, or we will be getting questions from users as to why we are showing a salinity that they haven't configured anywhere on any of their dive computers - or potentially show 'undefined' for salinity if this is the case.

Reply:
Well... Once I found function "fixup_water_salinity ", I thought that was a good solution for the issue...

> So I think if we display a calculated average for salinity

I did tests to understand it and, when we have more than 1 DC with salinity, the value showed is the result of fixup_water_salinity. So I conclude the salinity's average is used today and your concern is already real.
![image](https://user-images.githubusercontent.com/3233877/229011398-4b7f6074-00af-46c9-9b71-3765f4ccfa6c.png)
DC 1: `<water salinity='1030 g/l' />`
DC 2: `<water salinity='1040 g/l' />`

However, once it is used (and users know that), I believe this approach still the best at moment and we can open for discussion to find a better. Or we find a way to "mix" salinities ou we should to use just one of them, from current DC or from main one

Reply:
Yes, you are right. What you are doing is merely fixing the checks around the displayed salinity to be based on what is displayed and not around the current dive computer's reported salinity. The current behaviour seems to be wrong no matter what, and you are fixing it.